### PR TITLE
Refactor for SignalManager changes

### PR DESCRIPTION
**Pull request by williamsyang-work**
Replaces Signal keys with strings for event handlers Replaces fire* methods with emit methods

### CODE REVIEW

**Comment by williamsyang-work** [comment]
@ngondalia I'd like to know if the new typing inside `ExternalAPI` makes passes the type-enforcement into downstream repos.  LMK how it works.

File: vscode-trace-extension/src/trace-explorer/properties/trace-explorer-properties-view-webview-provider.ts [hunk 1]
```diff
@@ -33,26 +33,21 @@ export class TraceExplorerItemPropertiesProvider extends AbstractTraceExplorerPr
...
-    handleExperimentChanged = (exp: Experiment) => {
-        const props = this.propertiesMap.get(exp?.UUID);
-        if (props) {
-            this.handleUpdatedProperties(props);
-        } else {
-            const emptyPayload = new ItemPropertiesSignalPayload({});
-            this.handleUpdatedProperties(emptyPayload);
-        }
+    handleExperimentChanged = (exp?: Experiment) => {
+        const payload = exp ? this.propertiesMap.get(exp.UUID) : new ItemPropertiesSignalPayload({});
```
Comment (ngondalia) [inline vscode-trace-extension/src/trace-explorer/properties/trace-explorer-properties-view-webview-provider.ts:43]: I am getting 'TypeError: Cannot read properties of undefined (reading 'getExperimentUUID')'

The payload could be undefined here - Either have to address it here or could handle it in handleUpdatedProperties

Comment (williamsyang-work) [reply vscode-trace-extension/src/trace-explorer/properties/trace-explorer-properties-view-webview-provider.ts]: Yea, there was a weird issue where when `exp` is `Experiment | undfined` you can't pass `string | undefined` into the getter.  So i changed the logic but made some the assumption we would always find something if exp was defined..

Anyway... I added another checker so if `this.propertiesMap.get(exp.UUID)` doesn't find anything we just put an empty `ItemPropertiesSignalPayload` which should fix that.

**Comment by bhufmann** [comment]
@williamsyang-work could you please rebase this PR and fix the linter issue. There is a merge conflict with a recent change. Meanwhile I re-review the corresponding theia-trace-extension PR.

**Comment by williamsyang-work** [comment]
> @williamsyang-work could you please rebase this PR and fix the linter issues. There is a merge conflict with a recent change. Meanwhile I re-review the corresponding theia-trace-extension PR.

Sure.  I had this as a draft until the theia-extension PR was merged.  I will tidy this one up for proper review.

**Review by bhufmann — CHANGES_REQUESTED** [review]
The trace-viewer components have been released with version v0.5.0. (https://github.com/eclipse-cdt-cloud/theia-trace-extension/pull/1159). Please update this PR to upgrade to that version. Rrefer also #288 to see which files to update for the upgrade.

**Comment by bhufmann** [comment]
The following IP tickets have been created for the upgrade to traceviewer components v0.5.0:
- https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/17622
- https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/17623

**Review by bhufmann — APPROVED** [review]
It looks good to me. Thanks!

**Review by bhufmann — CHANGES_REQUESTED** [review]
I just realized that the README.md is describing the external API  and it needs to be updated. Please go throught the following chapter and update it the new API. The API is shown there and there is an example for the experiment selected signal as far I can see.

https://github.com/eclipse-cdt-cloud/vscode-trace-extension/blob/master/README.md#using-the-external-api

**Review by bhufmann — APPROVED** [review]
Looks good to me. Thanks!

**Comment by marcdumais-work** [comment]
There's an extra dependency requiring an IP Ticket that popped-up (it was not mentioned in the previous CI run):

image

IP ticket:
https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/17767

**Review by marcdumais-work — APPROVED** [review]
LGTM! Really nice contribution @williamsyang-work! (specially when considering the related ground work you first did in the theia-trace-extension repo)

**Comment by marcdumais-work** [comment]
> There's an extra dependency requiring an IP Ticket that popped-up (it was not mentioned in the previous CI run):
> 
> image
> 
> IP ticket: https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/17767

I do not think we need for this ticket to be resolved before merging - I had a look and I am 99.9% sure this License check failure is a false positive.

**Comment by bhufmann** [comment]
I'm going to merge this change. Then, I plan to do a release this week. @ngondalia please let me know if you have any concern about the release.

**Comment by marcdumais-work** [comment]
> > There's an extra dependency requiring an IP Ticket that popped-up (it was not mentioned in the previous CI run):
> > IP ticket: https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/17767


IP Ticket closed (approved)

**Comment by ngondalia** [comment]
> I'm going to merge this change. Then, I plan to do a release this week. @ngondalia please let me know if you have any concern about the release.

Thanks for the heads up! I will do sanity testing on our end and will get back to you.

**Comment by ngondalia** [comment]
> > I'm going to merge this change. Then, I plan to do a release this week. @ngondalia please let me know if you have any concern about the release.
> 
> Thanks for the heads up! I will do sanity testing on our end and will get back to you.

Hi Bernd, apologies for the delay. No issues from our end! Thanks :)

**Comment by bhufmann** [comment]
> > > I'm going to merge this change. Then, I plan to do a release this week. @ngondalia please let me know if you have any concern about the release.
> > 
> > 
> > Thanks for the heads up! I will do sanity testing on our end and will get back to you.
> 
> Hi Bernd, apologies for the delay. No issues from our end! Thanks :)

Great. I'll go ahead then and prepare a release. Thanks!

**Pull request merged**